### PR TITLE
Export Tally Data to VTK capability

### PR DESCRIPTION
**Pull request by pshriwise**
This PR adds functionality for exporting the current view of tally data to a VTK image file (.vti) for 3D visualization of tally results.

For tallies with mesh filters, the bounds and resolution of the VTK file is predetermined by the tool and cannot be altered by the user. 

For spatial domain filters, the user will specify min/max bounds for x, y, and z as well as the resolution in each dimension for the image file.

If the `vtk` module is not installed, then this feature is disabled.

For large datasets (big meshes or a fine resolution specified by the userr), a progress dialog will appear so the user knows something is happening. Additionally, domain information (cell/material ids) can be exported along with the tally info.

Other updates:

  - Image generation functions for tally data have been moved into the plotmodel file. That way all image generation (for cell/material ids, temp/density info, and tally data) all live in the same module.

  - Selected filter bins, scores, and nuclides have now been added to the model so the dock doesn't need to query widgets for the selected values. This allows for better separation of the GUI info and model.

  - Another set of changes from `FM` to `font_metric`.

  - Updated format for global variables related to tally info (as suggested in #33)

### CODE REVIEW

File: tools.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            self.xmaxBox.setEnabled(False)
+            self.xmaxBox.setToolTip(bounds_msg)
+            self.yminBox.setEnabled(False)
+            self.yminBox.setToolTip(bounds_msg)
+            self.ymaxBox.setEnabled(False)
+            self.ymaxBox.setToolTip(bounds_msg)
+            self.zminBox.setEnabled(False)
+            self.zminBox.setToolTip(bounds_msg)
+            self.zmaxBox.setEnabled(False)
+            self.zmaxBox.setToolTip(bounds_msg)
```
Comment (paulromano) [inline tools.py]:
```
for box in ('xminBox', 'xmaxBox', 'yminBox', 'ymaxBox', 'zminBox', 'zmaxBox'):
    getattr(self, box).setEnabled(False)
    getattr(self, box).setToolTip(bounds_msg)
```
Less repetitive, but maybe less clear? I'll leave it up to you if you want to adopt this.

File: tools.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        if any(llc >= urc):
+            self._warn("Bounds of export data are invalid.")
+            return
+
+        filename, ext = QtWidgets.QFileDialog.getSaveFileName(
+            self,
+            "Set VTK Filename",
+            "tally_data.vti",
+            "VTK Image (.vti)")
+        print(ext)
```
Comment (paulromano) [inline tools.py]: Remove this

File: tools.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        z0 = llc[2] + dz / 2.0
+
+        progressBar = QtWidgets.QProgressDialog("Accumulating data...",
+                                                "Abort Copy",
+                                                0,
+                                                res[2])
+        progressBar.setWindowModality(QtCore.Qt.WindowModal)
+
+        for k in range(res[2]):
+            z = z0 + k * dz
```
Comment (paulromano) [inline tools.py]:
```suggestion
            z = z0 + k*dz
```

File: tools.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+            view.origin = (x0, y0, z)
+            view.basis = 'xy'
+            self.model.activeView = view
+            self.model.makePlot()
+            image_data = self.model.create_tally_image(view)
+            data[k, :,:] = image_data[0][::-1, :]
```
Comment (paulromano) [inline tools.py]:
```suggestion
            data[k] = image_data[0][::-1, :]
```

File: tools.py [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            if include_geom:
+                mats[k, :, :] = self.model.mat_ids[::-1, :]
```
Comment (paulromano) [inline tools.py]:
```suggestion
                mats[k] = self.model.mat_ids[::-1, :]
```

File: tools.py [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+                cells[k, :, :] = self.model.cell_ids[::-1, :]
```
Comment (paulromano) [inline tools.py]:
```suggestion
                cells[k] = self.model.cell_ids[::-1, :]
```

File: docks.py [hunk 7]
```diff
@@ -1,53 +1,20 @@
...
-score_units['kappa-fission'] = energy_units
-score_units['fission-q-prompt'] = energy_units
-score_units['fission-q-recoverable'] = energy_units
-score_units['decay-rate'] = 's^-1'
-score_units['damage-energy'] = energy_units
-
-tally_values = {'Mean': 'mean',
-                'Std. Dev.': 'std_dev',
-                'Rel. Error': 'rel_err'}
+from plotmodel import (_SCORE_UNITS, _TALLY_VALUES,
```
[the remark this answers is not in the document]
Comment (pshriwise) [reply docks.py]: Missed that in my rebase. Thanks!

File: openmc-plotter [hunk 8]
```diff
@@ -359,6 +365,18 @@ class MainWindow(QMainWindow):
...
+        # Tools Menu
+        self.exportDataAction = QAction('E&xport Tally Data', self)
+        self.exportDataAction.setToolTip('Export current tally data to VTK')
+        self.setStatusTip('Export current tallly data to VTK')
+        self.exportDataAction.triggered.connect(self.exportTallyData)
+        if not _HAVE_VTK:
+            self.exportDataAction.setEnabled(False)
+            self.exportDataAction.setToolTip("Disabled: VTK Python module is not installed")
+
+        self.toolsMenu = self.mainMenu.addMenu('&Tools')
```
[the remark this answers is not in the document]
Comment (pshriwise) [reply openmc-plotter]: Good point. It is creating a file after all.

**Pull request merged**